File: app/src/main/java/com/jat/medilinkapp/retro_conf/ApiUtils.java
package com.jat.medilinkapp.retro_conf;

public class ApiUtils {
    private ApiUtils() {
    }

    //public static final String BASE_URL = "http://192.168.43.2:8080/medilinkWS/";
    public static final String BASE_URL = "http://%s/medilinkWS/";

    public static APIService getAPIService(String wsFromNfc) {
        //wsFromNfc = "216.114.104.227:21450";
        //wsFromNfc = "192.168.43.2:8080";

        String baseURL = String.format(BASE_URL, wsFromNfc);
        return RetrofitClient.getClient(baseURL).create(APIService.class);
    }
}

File: app/src/main/java/com/jat/medilinkapp/retro_conf/ApiUtils.kt
package com.jat.medilinkapp.retro_conf

object ApiUtils {
    val BASE_URL = "http://%s/medilinkWS/"

    val apiService: APIService
        get() {
            //val wsAddress = "216.114.104.227:21450"
            val wsAddress ="Medi-EVV.shepconnect.net:21450"
            //val wsAddress = "192.168.43.2:8080";
            val baseURL = String.format(BASE_URL, wsAddress)
            return RetrofitClient.getClient(baseURL)!!.create(APIService::class.java)
        }
}
